src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Attributes/JsonSerializableAttribute.cs

```diff
@@ -21,6 +21,11 @@ public sealed class JsonSerializableAttribute : JsonAttribute
         /// </remarks>
         public string? TypeInfoPropertyName { get; set; }
 
+        /// <summary>
+        /// Determines what the source generator should generate for the type.
+        /// </summary>
+        public JsonSourceGenerationMode GenerationMode { get; set; }
+
         /// <summary>
```

Comment:
Two existing issues in this attribute.

1. It is odd that the ctor is below the properties.
2. We don't store the `Type` ctor arg in a property anywhere?

Reply:
I've moved the ctor up & we don't need to store the `Type` ctor arg since it's for design-time inspection only and we can read the value directly from C# syntax metadata.